README.md
Techplaya
=============

This is the website that drives [techplaya.com](http://techplaya.com/). It's a showcase of people who work in dev/design/biz in Playa del Carmen. You might want to use it as a starting point for a showcase of your own community. Behind the scenes it uses [D3](http://d3js.com/), a really awesome javascript library for manipulating documents based on data. 

MIT License

Rationale
-------

Playa del Carmen is a small city along the Mexican Caribbean. It is a popular tourist destination and it hosts a small tech community. We would like to connect with other like-minded individuals who visit the area. 

Customization
-------

If you wish to customize techplaya for your own community you should modify these files:

* **src/index.html** Currently all information is hardcoded in HTML, you should modify this
* **src/js/destinations.js** Cities and airlines with direct flights
* **src/js/flights.js** `var location = [lat, long]`
* **src/js/satellite.js** displays the header map. Fiddle around with the projection to get the area of the world that you need.
* **src/js/satellite.js** specify the montly weather in Celsius.

Build
-------

`make release` will concatenate your JS and CSS and put it into the `release` directory. It will also modify your `index.html` to point to the minified files. You should deploy the files from this directory.

`make clean` will delete the release directory.


TODO
-------

* Put under a better build process such as middleman
* Create fallbacks for browsers that don't support SVG



CHANGES.md
0.2 - April 5, 2013
-------------------
* Build process included

0.1 - February 27, 2013
-------------------
* TechPlaya released


src/js/flights.js
techplaya = techplaya || {};
techplaya.flights = function(destinations){

  var world = techplaya.world;
  var destinations = techplaya.destinations;
  var location = [-86.85, 21.16];

  // FLIGHTS
  var width = 1200,
      height = 1000;

  var isTouch = !!('ontouchstart' in window) || !!('onmsgesturechange' in window);

  var flights_div = document.querySelector("section.flights");

  var projection = d3.geo.armadillo()
      .scale(500)
      .translate([width / 2, height*0.64])
      .parallel(20)
      .rotate([40, 0])
      .precision(.1)

  var path = d3.geo.path().projection(projection); 

  var graticule = d3.geo.graticule();


  var svg = d3.select("#map").append("svg")
      .attr("width", "100%")
      .attr("height", height)
      .attr("viewBox", "0 0 1200 1000");

  var defs = svg.append("defs");
  var gradient = defs.append("radialGradient")
  var colors = ["white", "white"]

  gradient
      .attr("r", "0.5")
      .attr("id", "world_gradient")
      .attr("gradientTransform", "translate(0, -0.5)")
      .selectAll("stop")
      .data(colors)
      .enter()
      .append("stop")
          .attr("offset",      function(d,i){return i})
          .attr("stop-opacity",function(d,i){return i == 0 ? 0.1 : 0})
          .attr("stop-color",  "white");

  var clipPath = defs.append("clipPath")
      .attr("id", "clip")
      .append("path")
          .datum({type: "Sphere"})
          .attr("id", "sphere")
          .attr("d", path);

  svg.append("use")
      .attr("class", "background")
      .attr("fill", "url(#world_gradient)")
      .attr("xlink:href", "#sphere");

  svg.append("g")
      .attr("class", "graticule")
      .attr("clip-path", "url(#clip)")
    .selectAll("path")
      .data(graticule.lines)
    .enter().append("path")
    .attr("d", path);

  svg.append("use")
     .attr("class", "foreground")
     .attr("xlink:href", "#sphere");

  svg.append("path", ".graticule")
    .datum(topojson.object(world, world.objects.land))
    .attr("class", "land")
    .attr("clip-path", "url(#clip)")
    .attr("d", path);

  svg.append("path", ".graticule")
    .datum(topojson.mesh(world, world.objects.countries, function(a, b) { return a !== b; }))
    .attr("class", "boundary")
    .attr("clip-path", "url(#clip)")
    .attr("d", path);


  //routes
  var route_color = d3.range(destinations.length).map(function(d, i) { return d3.hcl(i*2, 70,Math.random()*50+50).toString(); });
  var arc = d3.geo.greatArc()
      .source(function(d){return d.coords})
      .target(location)
      .precision(.1);

  var destinations = d3.shuffle(destinations)
  var routes = svg.selectAll("path.route")
      .data(destinations)
      .enter().append("path")
      .attr("data-city", function(d){return d.name})
      .attr("d", function(d) { return path(arc(d)); })
      .attr("stroke", function(d,i){return route_color[i]}) 
      .attr("stroke-dasharray", function(d){ return this.getTotalLength() + " " + this.getTotalLength() })
      .attr("stroke-dashoffset", function(d){ return isTouch ? 0 : this.getTotalLength()})
      .classed("route", true)

  var destination_el = d3.select("#destination");

  var cities = svg.selectAll("circle")
      .data(destinations)
      .enter().append("circle")
          .attr("class", function(d){return "city"})
          .attr("data-city", function(d){return d.name})
          .attr("cx", function(d){return projection(d.coords)[0]})
          .attr("cy", function(d){return projection(d.coords)[1]})
          .attr("fill", function(d,i){return route_color[i]}) 
          .attr("r", isTouch ? 15 : 1)
          .style("opacity", isTouch ? 0.3 : 1)
          .on("mouseover", showInfo)
          .on("touchstart", showInfo)
          .on("mouseout", hideInfo)

          function showInfo(d, i){
            d3.selectAll("[data-city='" + d.name + "']").style("opacity", 1)
            destination_el
              .style("border-color", route_color[i])
              .style("opacity", 1)
              .select("h2")
                .text(d.name)
                .style("color", route_color[i]);
            destination_el.select("#duration").text(d.flight_time);
            destination_el.select("#airlines").text(d.airlines.join(", "));
          }

          function hideInfo(d){
              d3.selectAll("[data-city='" + d.name + "']")
                  .transition()
                  .duration(300)
                  .style("opacity", 0.3)
          };

  window.onscroll = isTouch ? null : function(){ techplaya.throttle(routes_animation, window)};

  function routes_animation(){
    if (window.scrollY < flights_div.offsetTop - flights_div.offsetHeight) return;
    window.onscroll = null;
    radar = null;
    cities.transition()
      .delay(function(d, i){return i*20})
      .attr("r", 10)
      .style("opacity", 0.3)
      .ease("out")
      .each("end", function(){d3.select(this).transition().ease("in").attr("r", 4)});

    routes.transition()
      .delay(function(d, i){return i*20 + 500})
      .duration(3000)
      .attr("stroke-dashoffset", 0);
  }
};

src/js/satellite.js
techplaya.satellite = function(){

    var world = techplaya.world;

    //satellite projection for header

    var width = 1600,
        height = 600;
     
    var projection = d3.geo.satellite()
        .distance(1.7)
        .scale(2500)
        .rotate([90.00, -5, 0])
        .center([-10, 10])
        .tilt(0)
        .clipAngle(50);

    var graticule = d3.geo.graticule()
        .extent([[-150, 10], [-30 + 1e-6, 57]])
        .step([3, 3]);
     
    var path = d3.geo.path()
        .projection(projection);
     
    var svg = d3.select("#satellite").append("svg")
        .attr("viewBox", "0 0 " + width + " " + height);

    svg.append("rect")
        .attr("x", -5)
        .attr("y", -5000)
        .attr("width", width)
        .attr("height", 10000)
        .attr("fill", "black");
        
    var defs = svg.append("defs");

    var clipPath = defs.append("clipPath")
        .attr("id", "clip-intro")
        .append("path")
            .datum({type: "Sphere"})
            .attr("id", "sphere-intro")
            .attr("d", path);

    svg.append("use")
       .attr("class", "foreground")
       .attr("xlink:href", "#sphere-intro");

    svg.append("path")
        .datum(graticule)
        .attr("class", "graticule")
        .style("opacity", "0.3")
        .attr("clip-path", "url(#clip-intro)")
        .attr("d", path)
     
    svg.append("path")
        .datum(topojson.object(world, world.objects.land))
        .attr("class", "land")
        .attr("d", path);

    svg.append("circle")
        .attr("class", "city")
        .attr("cx", projection([-87.06, 20.6167])[0])
        .attr("cy", projection([-87.06, 20.6167])[1])
        .attr("r", 5)

    var home_city = svg.append("circle")
        .attr("cx", projection([-87.06, 20.6167])[0])
        .attr("cy", projection([-87.06, 20.6167])[1])
        .attr("fill", "none")
        .attr("stroke", "black")

    function radar(){
      home_city
        .attr("r", 10)
        .attr("stroke-width", 5)
        .attr("stroke-opacity", 0.3)
        .transition()
        .duration(1000)
          .attr("r", 20)
          .attr("stroke-width", 10)
          .attr("stroke-opacity", 0)
          .each("end", radar)
    }

    var satellite  = d3.select("#satellite svg");
    function map_adjust(){
      var proportion = 1600 / window.innerWidth * 600
      satellite.attr("viewBox", "0 -" + proportion/2 + " 1600 " + 1600 / window.innerWidth * 600);
    }

    window.onresize = map_adjust; 

    map_adjust();
    radar();

};




src/js/destinations.js
techplaya.destinations = [
    {
        name: "Moscow",
        coords: [37.6178,55.7517],
        airlines: ["Aeroflot"],
        flight_time: "14:57"
    },
    {
        name: "Buenos Aires",
        coords: [-58.3817,-34.6036],
        airlines: ["Aerolíneas Argentinas"],
        flight_time: "9:00"
    },
    {
        name: "Bogotá",
        coords: [-74.07,4.59],
        airlines: ["Aeroméxico", "Avianca", "Copa Airlines Colombia"],
        flight_time: "3:21"
    },
    {
        name: "Havana",
        coords: [-82.39,23.12],
        airlines: ["Aeroméxico", "Cubana"],
        flight_time: "1:08"
    },
    {
        name: "Mexico City",
        coords: [-99.40,19.13],
        airlines: ["Aeroméxico", "Interjet", "Magincharters", "VivaAerobus", "Volaris"],
        flight_time: "2:07"
    },
    {
        name: "San José",
        coords: [-84.08,9.92],
        airlines: ["Aeroméxico"],
        flight_time: "1:38"
    },
    {
        name: "New York",
        coords: [-74.00,40.71],
        airlines: ["Aeroméxico", "United Airways", "American Airlines", "Jet Blue", "Allegiant Airlines"],
        flight_time: "3:00"
    },
    {
        name: "Monterrey",
        coords: [-100.30,25.67],
        airlines: ["Aeroméxico Connect", "Interjet", "Viva Aerobus"],
        flight_time: "2:10"
    },
    {
        name: "Cozumel",
        coords: [-86.95,20.50],
        airlines: ["Maya Air", "Aerotucán", "Viva Aerobus"],
        flight_time: "00:20"
    },
    {
        name: "Düsseldorf",
        coords: [6.78,51.23],
        airlines: ["Air Berlin"],
        flight_time: "11:00"
    },
    {
        name: "Munich",
        coords: [11.57,48.13],
        airlines: ["Air Berlin"],
        flight_time: "11:32"
    },
    {
        name: "Calgary",
        coords: [-114.07, 51.05],
        airlines: ["Air Canada", "Air Transat", "CanJet", "Enerjet", "WestJet"],
        flight_time: "5:34"
    },
    {
        name: "Montréal",
        coords: [-73.56, 45.51],
        airlines: ["Air Canada", "Air Transat", "CanJet", "Sunwing Airlines", "WestJet"],
        flight_time: "4:11"
    },
    {
        name: "Toronto",
        coords: [-79.40, 43.65],
        airlines: ["Air Canada", "Air Transat", "CanJet", "Enerjet", "Sunwing Airlines", "WestJet"],
        flight_time: "3:43"
    },
    {
        name: "Halifax",
        coords: [-63.57, 44.65],
        airlines: ["Air Canada", "Air Transat", "CanJet", "Sunwing Airlines", "WestJet"],
        flight_time: "4:41"
    },
    {
        name: "Ottawa",
        coords: [-75.69, 45.42],
        airlines: ["Air Canada", "CanJet", "WestJet"],
        flight_time: "4:04"
    },
    {
        name: "Winnipeg",
        coords: [-97.15, 49.88],
        airlines: ["Air Canada", "CanJet", "Sunwing Airlines", "WestJet"],
        flight_time: "4:37"
    },
    {
        name: "Madrid",
        coords: [-3.68, 40.40],
        airlines: ["Air Europa", "Air Pullmantur"],
        flight_time: "10:30"
    },
    {
        name: "Edmonton",
        coords: [-113.50, 53.55],
        airlines: ["AirTransat","CanJet", "Sunwing Airlines", "WestJet", "Enerjet"],
        flight_time: "5:47"
    },
    {
        name: "Québec",
        coords: [-71.22, 46.89],
        airlines: ["AirTransat","CanJet", "Sunwing Airlines", "WestJet"],
        flight_time: "4:28"
    },
    {
        name: "Regina",
        coords: [-104.61, 50.46],
        airlines: ["AirTransat","CanJet", "Sunwing Airlines", "WestJet"],
        flight_time: "5:00"
    },
    {
        name: "Saskatoon",
        coords: [-106.64, 52.13],
        airlines: ["AirTransat", "Sunwing Airlines", "WestJet"],
        flight_time: "5:06"
    },
    {
        name: "Atlanta",
        coords: [-84.39, 33.75],
        airlines: ["Southwest Airlines", "Delta Airlines"],
        flight_time: "2:16"
    },
    {
        name: "Austin",
        coords: [-97.74, 30.27],
        airlines: ["Southwest Airlines", "United Airlines"],
        flight_time: "2:21"
    },
    {
        name: "Baltimore",
        coords: [-76.62, 39.28],
        airlines: ["Southwest Airlines"],
        flight_time: "3:17"
    },
    {
        name: "Chicago",
        coords: [-87.65, 41.85],
        airlines: ["Southwest Airlines", "United Airlines", "American Airlines", "Frontier Airlines"],
        flight_time: "3:21"
    },
    {
        name: "Denver",
        coords: [-104.98, 39.74],
        airlines: ["Southwest Airlines", "Frontier Airlines", "United Airlines"],
        flight_time: "3:50"
    },
    {
        name: "San Antonio",
        coords: [-98.49, 29.42],
        airlines: ["Southwest Airlines", "United Airlines"],
        flight_time: "2:21"
    },
    {
        name: "Milwaukee",
        coords: [-87.91, 43.04],
        airlines: ["Southwest Airlines", "Delta Airlines", "Frontier Airlines"],
        flight_time: "3:31"
    },
    {
        name: "Dallas",
        coords: [-96.80, 32.78],
        airlines: ["American Airlines", "Spirit Airlines", "Sun Country Airlines"],
        flight_time: "2:31"
    },
    {
        name: "Miami",
        coords: [-80.13, 25.79],
        airlines: ["American Airlines"],
        flight_time: "1:57"
    },
    {
        name: "Detroit",
        coords: [-83.05, 42.33],
        airlines: ["Airtran Airways", "Delta Airlines", "Spirit Airlines"],
        flight_time: "1:33"
    },
    {
        name: "Philadelphia",
        coords: [-75.16, 39.95],
        airlines: ["US Airways", "Frontier Airlines"],
        flight_time: "3:26"
    },
    {
        name: "Amsterdam",
        coords: [4.89, 52.37],
        airlines: ["Arkefly"],
        flight_time: "10:48"
    },
    {
        name: "Bologna",
        coords: [11.35, 44.50],
        airlines: ["Blue Panorama Airlines"],
        flight_time: "11:42"
    },
    {
        name: "Milan",
        coords: [9.19, 45.46],
        airlines: ["Blue Panorama Airlines"],
        flight_time: "11:27"
    },
    {
        name: "Rome",
        coords: [12.50, 41.90],
        airlines: ["Blue Panorama Airlines"],
        flight_time: "11:56"
    },
    {
        name: "London",
        coords: [0.11, 51.52],
        airlines: ["British Airways", "Monarch Airlines", "Tomas Cook Airlines", "Thomson Airways", "Virgin Atlantic Airways"],
        flight_time: "10:24"
    },
    {
        name: "London, Ontario",
        coords: [-81.15, 43.03],
        airlines: ["CanJet", "Sunwing Airlines", "WestJet"],
        flight_time: "3:35"
    },
    {
        name: "Vancouver",
        coords: [-123.11, 49.25],
        airlines: ["Airtransat",  "CanJet", "Sunwing Airlines", "WestJet", "Enerjet"],
        flight_time: "6:03"
    },
    {
        name: "Hamilton",
        coords: [-79.87, 43.26],
        airlines: ["CanJet", "WestJet"],
        flight_time: "3:39"
    },
    {
        name: "Kelowna",
        coords: [-119.49, 49.88],
        airlines: ["CanJet", "WestJet"],
        flight_time: "5:50"
    },
    {
        name: "Victoria",
        coords: [-123.33, 48.43],
        airlines: ["CanJet", "WestJet"],
        flight_time: "6:01"
    },
    {
        name: "Frankfurt",
        coords: [8.68, 50.12],
        airlines: ["Condor"],
        flight_time: "11:12"
    },
    {
        name: "Paris",
        coords: [2.35, 48.87],
        airlines: ["Air France", "Corsair International", "XL Airways France"],
        flight_time: "10:42"
    },
    {
        name: "Panamá City",
        coords: [-79.50, 9],
        airlines: ["Copa Airlines"],
        flight_time: "2:26"
    },
    {
        name: "Cincinnati",
        coords: [-84.46, 39.16],
        airlines: ["Delta Airlines", "Fontier Airlines"],
        flight_time: "3:00"
    },
    {
        name: "Minneapolis",
        coords: [-93.26, 44.98],
        airlines: ["Delta Airlines", "Sun Country Airlines"],
        flight_time: "3:52"
    },
    {
        name: "Boston",
        coords: [-71.06, 42.36],
        airlines: ["Delta Airlines", "JetBlue Airways", "US Airways"],
        flight_time: "3:57"
    },
    {
        name: "Columbus",
        coords: [-83.00, 39.97],
        airlines: ["Delta Airlines"],
        flight_time: "3:08"
    },
    {
        name: "Hartford/Springfield",
        coords: [-72.68, 41.76],
        airlines: ["Delta Airlines"],
        flight_time: "3:47"
    },
    {
        name: "Indianapolis",
        coords: [-86.16, 39.77],
        airlines: ["Delta Airlines", "Frontier Airlines"],
        flight_time: "3:04"
    },
    {
        name: "Los Angeles",
        coords: [-118.24, 34.05],
        airlines: ["United Airlines", "Virgin America"],
        flight_time: "4:43"
    },
    {
        name: "Memphis",
        coords: [-90.05, 35.15],
        airlines: ["Delta Airlines"],
        flight_time: "2:28"
    },
    {
        name: "Nashville",
        coords: [-86.78, 36.17],
        airlines: ["Delta Airlines", "Aeroméxico"],
        flight_time: "2:34"
    },
    {
        name: "Orlando",
        coords: [-81.38, 28.54],
        airlines: ["Delta Airlines", "JetBlue Airways"],
        flight_time: "1:43"
    },
    {
        name: "Raleigh-Durham",
        coords: [-78.64, 35.77],
        airlines: ["Delta Airlines", "JetBlue Airways"],
        flight_time: "2:45"
    },
    {
        name: "Salt Lake City",
        coords: [-111.88, 40.75],
        airlines: ["Delta Airlines", "Frontier Airlines"],
        flight_time: "4:29"
    },
    {
        name: "Tampa",
        coords: [-82.46, 27.95],
        airlines: ["Delta Airlines"],
        flight_time: "1:35"
    },
    {
        name: "Washington",
        coords: [-77.03, 38.89],
        airlines: ["Delta Airlines", "United Airlines"],
        flight_time: "3:12"
    },
    {
        name: "Zürich",
        coords: [8.54, 47.37],
        airlines: ["Edelweiss Air"],
        flight_time: "11:18"
    },
    {
        name: "Lisbon",
        coords: [-9.18, 38.70],
        airlines: ["EuroAtlantic Airways", "Orbest", "White"],
        flight_time: "9:51"
    },
    {
        name: "Saint Petersburg",
        coords: [30.33, 59.93],
        airlines: ["Orenair"],
        flight_time: "12:10"
    },
    {
        name: "Porlamar",
        coords: [-63.85, 10.95],
        airlines: ["Perla Airlines"],
        flight_time: "3:51"
    },
    {
        name: "Fort Lauderdale",
        coords: [-80.14, 26.12],
        airlines: ["Spirit Airlines"],
        flight_time: "1:35"
    },
    {
        name: "Lansing",
        coords: [-84.56, 42.73],
        airlines: ["Sun Country Airlines"],
        flight_time: "3:29"
    },
    {
        name: "Saguenay",
        coords: [-71.08, 48.43],
        airlines: ["Sunwing"],
        flight_time: "4:19"
    },
    {
        name: "Sault Ste. Marie",
        coords: [-84.34, 46.52],
        airlines: ["Sunwing Airlines"],
        flight_time: "4:01"
    },

    {
        name: "San Salvador",
        coords: [-89.18, 13.68],
        airlines: ["TACA Airlines"],
        flight_time: "1:33"
    },
    {
        name: "Manchester",
        coords: [-2.24, 53.48],
        airlines: ["Monarch Airlines", "Thomson Airlines", "Thomas Cook Airlines"],
        flight_time: "10:12"
    },
    {
        name: "Glasgow",
        coords: [-4.27, 55.87],
        airlines: ["Thomson Airlines", "Thomas Cook Airlines"],
        flight_time: "10:00"
    },
    {
        name: "Belfast",
        coords: [-5.92, 54.60],
        airlines: ["Thomas Cook Airlines"],
        flight_time: "9:53"
    },
    {
        name: "Birmingham",
        coords: [-1.91, 52.48],
        airlines: ["Thomson Airways"],
        flight_time: "10:15"
    },
    {
        name: "Bristol",
        coords: [-2.60, 51.46],
        airlines: ["Thomson Airways"],
        flight_time: "10:13"
    },
    {
        name: "East Midlands",
        coords: [-1.33, 52.83],
        airlines: ["Thomson Airways"],
        flight_time: "10:15"
    },
    {
        name: "Edinburg",
        coords: [-3.22, 55.95],
        airlines: ["Thomson Airways"],
        flight_time: "10:05"
    },
    {
        name: "Newcastle upon Tyne",
        coords: [-1.58, 54.98],
        airlines: ["Thomson Airways"],
        flight_time: "10:13"
    },
    {
        name: "Flores",
        coords: [-89.90, 16.90],
        airlines: ["Transportes Aereos Guatemaltecos"],
        flight_time: "1:11"
    },
    {
        name: "Copenhagen",
        coords: [12.57, 55.68],
        airlines: ["TUIfly Nordic"],
        flight_time: "11:17"
    },
    {
        name: "Gothenburg-Landvetter",
        coords: [12.28, 57.66],
        airlines: ["TUIfly Nordic"],
        flight_time: "11:8"
    },
    {
        name: "Oslo",
        coords: [10.76, 59.95],
        airlines: ["TUIfly Nordic"],
        flight_time: "10:58"
    },
    {
        name: "Stockholm",
        coords: [18.07, 59.33],
        airlines: ["TUIfly Nordic"],
        flight_time: "11:28"
    },
    {
        name: "Cleveland",
        coords: [-81.70, 41.50],
        airlines: ["United Airlines"],
        flight_time: "3:23"
    },
    {
        name: "Houston",
        coords: [-95.36, 29.76],
        airlines: ["United Airlines"],
        flight_time: "2:06"
    },
    {
        name: "San Francisco",
        coords: [-122.42, 37.78],
        airlines: ["United Airlines", "Virgin America"],
        flight_time: "5:20"
    },
    {
        name: "Cuernavaca",
        coords: [-99.23, 18.92],
        airlines: ["VivaAerobus"],
        flight_time: "2:08"
    },
    {
        name: "Guadalajara",
        coords: [-103.35, 20.67],
        airlines: ["VivaAerobus", "Volaris"],
        flight_time: "2:38"
    },
    {
        name: "Tuxtla Gutiérrez",
        coords: [-93.12, 16.75],
        airlines: ["VivaAerobus"],
        flight_time: "1:30"
    },
    {
        name: "Veracruz",
        coords: [-96.14, 19.19],
        airlines: ["Maya Air", "VivaAerobus"],
        flight_time: "1:44"
    },
    {
        name: "Santiago de Chile",
        coords: [-70.67, -33.45],
        airlines: ["LAN Airlines"],
        flight_time: "8:18"
    },
    {
        name: "Lima",
        coords: [-77.03, -12.04],
        airlines: ["LAN Peru"],
        flight_time: "5:14"
    },
    {
        name: "Belize City",
        coords: [-88.18, 17.50],
        airlines: ["Maya Island Air"],
        flight_time: "1:01"
    },
    {
        name: "Ciudad del Carmen",
        coords: [-91.83, 18.63],
        airlines: ["Maya Air"],
        flight_time: "1:00"
    },
    {
        name: "Mérida",
        coords: [-89.62, 20.98],
        airlines: ["Maya Air"],
        flight_time: "0:51"
    },
    {
        name: "Villahermosa",
        coords: [-92.59, 17.97],
        airlines: ["Maya Air", "VivaAerobus"],
        flight_time: "1:24"
    },
    {
        name: "Aguascalientes",
        coords: [-102.36, 22.02],
        airlines: ["Volaris"],
        flight_time: "2:29"
    },
    {
        name: "Tijuana",
        coords: [-117.02, 32.53],
        airlines: ["Volaris"],
        flight_time: "4:32"
    },
    {
        name: "León",
        coords: [-101.68, 21.12],
        airlines: ["Volaris"],
        flight_time: "2:25"
    },
    {
        name: "Puebla",
        coords: [-97.89, 19.00],
        airlines: ["Volaris"],
        flight_time: "1:30"
    },
    {
        name: "Querétaro",
        coords: [-99.85, 20.84],
        airlines: ["Volaris"],
        flight_time: "2:15"
    },
    {
        name: "Toluca",
        coords: [-99.77, 19.10],
        airlines: ["Volaris"],
        flight_time: "2:11"
    },
    {
        name: "Moncton",
        coords: [-64.78, 46.09],
        airlines: ["WestJet"],
        flight_time: "4:46"
    },
    {
        name: "Thunder Bay",
        coords: [-89.25, 48.41],
        airlines: ["WestJet"],
        flight_time: "4:17"
    },
    {
        name: "Porto",
        coords: [-8.58, 41.17],
        airlines: ["White", "EuroAtlantic Airways"],
        flight_time: "9:52"
    },
    {
        name: "Pittsburg",
        coords: [-80.00, 40.44],
        airlines: ["Frontier Airlines"],
        flight_time: "3:32"
    },
    {
        name: "Kansas City",
        coords: [-94.58, 39.10],
        airlines: ["Frontier Airlines"],
        flight_time: "3:09"
    },
    {
        name: "Campinas-Viracopos",
        coords: [-47.06, -22.91],
        airlines: ["Whitejets"],
        flight_time: "8:35"
    },
    {
        name: "Rio de Janeiro",
        coords: [-43.24, -22.91],
        airlines: ["Whitejets"],
        flight_time: "8:56"
    },
    {
        name: "São Paulo",
        coords: [-46.62, -23.50],
        airlines: ["Whitejets", "Aeroméxico"],
        flight_time: "8:41"
    },
    {
        name: "Chetumal",
        coords: [-88.33, 18.50],
        airlines: ["Xiknal Air"],
        flight_time: "00:54"
    },
    {
        name: "Helsinki",
        coords: [24.94, 60.17],
        airlines: ["Finnair"],
        flight_time: "11:51"
    },
    {
        name: "Saint John",
        coords: [-66.06, 45.27],
        airlines: ["Sunwing airlines"],
        flight_time: "4:36"
    }

];

src/js/weather.js
techplaya.weather = function() {
  var celsius = [
    { month: "January",   high: 27, low: 20 },
    { month: "February",  high: 27, low: 20 },
    { month: "March",     high: 28, low: 21 },
    { month: "April",     high: 29, low: 22 },
    { month: "May",       high: 31, low: 23 },
    { month: "June",      high: 29, low: 24 },
    { month: "July",      high: 31, low: 24 },
    { month: "August",    high: 31, low: 24 },
    { month: "September", high: 31, low: 24 },
    { month: "October",   high: 28, low: 23 },
    { month: "November",  high: 27, low: 22 },
    { month: "December",  high: 27, low: 21 }
  ];

  var fahrenheit = celsius.map(function(d, i){
    function convert(temp) { return Math.round(temp * 9/5 + 32, 0) };
    return {
      high: convert(d.high),
      low: convert(d.low)
    }
  });

  var scale = d3.scale.linear()
    .domain([0, 50])
    .range([0, 100]);

  var w_div = d3.select("#weather");

  w_div.selectAll("div")
    .data(celsius)
    .enter()
    .append("div")
      .classed("month", true)
        .text(function(d){return d.month})
        .append("div")
          .classed("range", true)
          .style("left", function(d){return scale(d.low) + "%"})
          .style("width", function(d){return scale(d.high - d.low) + "%"})
          .attr("data-low", function(d){return d.low})
          .attr("data-high", function(d){return d.high})

  var units = d3.selectAll(".units .unit");

  units.on("click", switchUnits); 
  units.on("touchstart", switchUnits); 

  function switchUnits(){
    var isCelsius = d3.select(this).classed("celsius");
    units.classed("selected", false);
    d3.select(this).classed("selected", true);
    var data =  isCelsius ? celsius : fahrenheit;
    scale.domain(isCelsius ? [0, 50] : [0, 100])
    
    w_div.selectAll(".range")
      .style("left", "30%")
      .style("width", "70%");
    setTimeout(function(){
      w_div.selectAll(".range")
        .data(data)
        .style("left", function(d){return scale(d.low) + "%"})
        .style("width", function(d){return scale(d.high - d.low) + "%"})
        .attr("data-low", function(d){return d.low})
        .attr("data-high", function(d){return d.high})
    }, 500);
          
    w_div.classed("fahrenheit", !isCelsius);
  };
};